tests/admin-unlock/main.yml

```diff
@@ -43,338 +43,94 @@
-      tags:
-        - redhat_unsubscribe
+    - include_tasks: 'setup.yml'
+
+    # TEST
+    # Verify installed/removed packages do not persist through reboot with unlock
+    - block:
+        - include_tasks: 'add_remove_package_non_persistence.yml'
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'Add/Remove Package - Non Persistence', 'result':'Passed', 'result_details': '' } ] }}"
+      rescue:
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'Add/Remove Package - Non Persistence', 'result':'Failed', 'result_details': ansible_failed_result } ] }}"
+
+
+    # TEST
+    # Verify installed/removed packages persist through reboot with hotfix unlock
+    - block:
+        - include_tasks: 'add_remove_package_persistence.yml'
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'Add/Remove Package - Persistence', 'result':'Passed', 'result_details': '' } ] }}"
+      rescue:
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'Add/Remove Package - Persistence', 'result':'Failed', 'result_details': ansible_failed_result } ] }}"
+
+
+    # TEST
+    # Verify rollback to original deployment after hotfix unlock
+    - block:
+        - include_tasks: 'hotfix_rollback.yml'
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'Hotfix rollback', 'result':'Passed', 'result_details': '' } ] }}"
+      rescue:
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'Hotfix rollback', 'result':'Failed', 'result_details': ansible_failed_result } ] }}"
+
+
+    # TEST
+    # Verify no packages can be installed wihtout unlock
+    - block:
+        - include_tasks: 'no_install_without_unlock.yml'
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'No install without unlock', 'result':'Passed', 'result_details': '' } ] }}"
+      rescue:
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'No install without unlock', 'result':'Failed', 'result_details': ansible_failed_result } ] }}"
+
+
+    # TEST
+    # Verify unlocking twice causes error
+    - block:
+        - include_tasks: 'unlock_twice_error.yml'
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'Unlock twice error', 'result':'Passed', 'result_details': '' } ] }}"
+      rescue:
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'Unlock twice error', 'result':'Failed', 'result_details': ansible_failed_result } ] }}"
+
+
+    # TEST
+    # Verify hotfix unlocking twice causes error
+    - block:
+        - include_tasks: 'unlock_twice_hotfix_error.yml'
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'Unlock twice hotfix error', 'result':'Passed', 'result_details': '' } ] }}"
+      rescue:
+        - set_fact:
+            tests: "{{ tests + [ { 'name':'Unlock twice hotfix error', 'result':'Failed', 'result_details': ansible_failed_result } ] }}"
+
+    # CLEANUP
+    - block:
+        # WRITE RESULTS TO FILE
+        - name: Save result to file
```

Comment:
I think this should use `become: false`; since it is a local action, it shouldn't require privilege to write a file locally.